diff --git a/src/lammps.h b/src/lammps.h
index c432784..abfc8f4 100644
--- a/src/lammps.h
+++ b/src/lammps.h
@@ -16,6 +16,9 @@
 
 #include <stdio.h>
 
+// Global variable, makes life easier here
+extern class CPLSocketLAMMPS cplsocket;
+
 namespace LAMMPS_NS {
 
 class LAMMPS {
diff --git a/src/main.cpp b/src/main.cpp
index cc8f8be..c3c6bd7 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -17,20 +17,24 @@
 #include "error.h"
 #include <stdio.h>
 #include <stdlib.h>
+#include "CPLSocketLAMMPS.h"
 
 using namespace LAMMPS_NS;
 
 /* ----------------------------------------------------------------------
    main program to drive LAMMPS
 ------------------------------------------------------------------------- */
+CPLSocketLAMMPS cplsocket;
 
 int main(int argc, char **argv)
 {
   MPI_Init(&argc,&argv);
+  cplsocket.initComms();
+  MPI_Comm comm = cplsocket.realmCommunicator();
 
 #ifdef LAMMPS_EXCEPTIONS
   try {
-    LAMMPS *lammps = new LAMMPS(argc,argv,MPI_COMM_WORLD);
+    LAMMPS *lammps = new LAMMPS(argc,argv,comm);
     lammps->input->file();
     delete lammps;
   } catch(LAMMPSAbortException & ae) {
@@ -40,10 +44,10 @@ int main(int argc, char **argv)
     exit(1);
   }
 #else
-  LAMMPS *lammps = new LAMMPS(argc,argv,MPI_COMM_WORLD);
+  LAMMPS *lammps = new LAMMPS(argc,argv,comm);
   lammps->input->file();
   delete lammps;
 #endif
-  MPI_Barrier(MPI_COMM_WORLD);
+  MPI_Barrier(comm);
   MPI_Finalize();
 }
